clients/src/main/java/org/apache/kafka/clients/consumer/internals/StreamsGroupInitializeRequestManager.java

```diff
@@ -95,18 +100,65 @@ private static StreamsGroupInitializeRequestData.Subtopology getSubtopologyFromS
         final StreamsGroupInitializeRequestData.Subtopology subtopologyData = new StreamsGroupInitializeRequestData.Subtopology();
         subtopologyData.setSubtopologyId(subtopologyName);
         subtopologyData.setSourceTopics(new ArrayList<>(subtopology.sourceTopics));
+        Collections.sort(subtopologyData.sourceTopics());
```

Comment:
Why not
```java
final List<String> sortedSourceTopics = new ArrayList<>(subtopology.sourceTopics)
Collections.sort(sortedSourceTopics);
subtopologyData.setSourceTopics(sortedSourceTopics);
```
I think this would be easier to understand.

Reply:
Do you sort so that the requests are deterministic? I do not see any other reason at the moment.

Reply:
Yes. We want to make the topology deterministic, otherwise we'll get reinitializations